### DISCUSSION TITLE
Delisting or Removing Packages

### DISCUSSION THREAD

**Discussion by makma** [cdnjs/cdnjs#14062] [General]
Hello 👋,
I'd like to know how to delist or remove the existing package. I've read through the [contribution guidelines](https://github.com/cdnjs/packages/blob/master/CONTRIBUTING.md), but I haven't found any guidance about this topic. 
I assume removing `myPackage.json` from the `cdnjs/packages/packages/` might do the trick, but this leads me to other questions:
- What happens with the scripts that are served and used by users? Will they return 404 after my removal PR is merged?
- If so, is there a way to just delist a package but keep the current script served so we won't break users' projects?
- Is there a way to delist packages so they are no more visible on cdnjs.com and are not indexed by search engines?

Thank you.
Cheers! 🎉

**Comment by MattIPv4** [comment]
👋 The short answer is that we do not have a way to delist/remove packages -- once they are on the CDN, they are there forever, to ensure that users will always be able to access the assets, and the SRI data etc. for those assets via the website and CDN.

If a library is no longer being updated, we can remove the auto-update config from the package's JSON file in cdnjs/packages, but we cannot remove the JSON file itself, nor remove assets from the CDN or data from the API/website.

**Reply by makma** [reply]
Thank you for your answer, @MattIPv4. One more question - is there a way to add `noindex` meta tag for a given package if the owner of the library no more wants the package to be included in the search engine results?

**Reply by MattIPv4** [reply]
This isn't currently a feature that we have. I'm definitely not against the idea though, we've just never built it. Removing a library from our Algolia index would likely be relatively impossible, as this drives a lot of our API data which we don't want to ever be removing libraries from, but I don't see why we couldn't add a way to have a property in a package's JSON file that sets a noindex on the website.

**Reply by makma** [reply]
I understand it is in the nature of things that the package cannot be removed, nevertheless, not indexing the package would be very helpful. Can I file an issue from this discussion?

**Reply by MattIPv4** [reply]
Sure thing, would be best to create an issue in cdnjs/static-website for this request :)

**Reply by makma** [reply]
Great, thanks, will do.
Edit: [Issue](https://github.com/cdnjs/static-website/issues/89)
